src/workloads/query/GroupStagesOnComputedFields.yml

```diff
@@ -0,0 +1,176 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  The queries in this workload exercise group stage(s) after aggregation stages on computed date
+  fields.
+
+Keywords:
+- aggregate
+- sbe
+
+GlobalDefaults:
+  Database: &Database test
+  Collection: &Collection Collection0
+  MaxPhases: &MaxPhases 5
+
+MultipleGroupStagesOnComputedFields: &MultipleGroupStagesOnComputedFields
+  LoadConfig: &loadConfig
+    Path: "../../phases/query/GroupStagesOnComputedFields.yml"
+    Key: QueryTemplate
+    Parameters:
+      name: MultipleGroupStagesOnComputedFields
+      active: {^Parameter: {Name: "active", Default: [3]}}
+      nopInPhasesUpTo: &NopInPhasesUpTo {^Parameter: {Name: "nopInPhasesUpTo", Default: *MaxPhases}}
+      pipeline: [{$addFields:
+                          {"hourDiff": {$dateDiff: {"startDate": "$time", "endDate": {^Date: "2022-01-04"}, "unit": "hour"}}}},
+                          {$match: {"hourDiff": {$lte: 12}}},
+                          {$group:
+                            {"_id" :
+                              { "symbol" : "$symbol" ,
+                                "time": { "$dateTrunc" : {"date": "$time", "unit": "minute", "binSize" : 60}}
+                              },
+                              "open" : {"$first" : "$price"},
+                              "close" : {"$last" : "$price"}}
+                          },
+                          {
+                              $addFields : {
+                                  "diffPercentage" : {$round : [{$multiply : [100, { $divide : [{$subtract : ["$close", "$open"]}, "$open"] }]}, 2]}
+                              }
+                          },
+                          {
+                            $group: {
+                                  "_id" : "$_id.time",
+                                  "topN": {
+                                      $topN: {
+                                          "output": ["$_id.symbol", "$diffPercentage", "$close"],
+                                          "sortBy" : { "diffPercentage" : -1 },
+                                          "n": 3
+                                      }
+                                  },
+                                  "bottomN": {
+                                      $bottomN: {
+                                          "output": ["$_id.symbol", "$diffPercentage", "$close"],
+                                          "sortBy" : { "diffPercentage" : 1 },
+                                          "n": 3
+                                      }
+                                  },
+                              }
+                          },
+                          {
+                              $sort : {
+                                  "_id" : -1
+                              }
+                          }]
+
+GroupStageAfterMatchOnDateComputation: &GroupStageAfterMatchOnDateComputation
```

Comment:
I would rewrite this name to be `GroupStateOnDateComputationAfterMatch` so that it's clear that the date computation is in the group and not the match filter.

Reply:
Yes! Done.